inspirehep/modules/authors/views.py

```diff
@@ -230,21 +128,34 @@ def new():
 @login_required
 def update(recid):
     """View for INSPIRE author update form."""
-    data = {}
-    if recid:
+    try:
+        data = get_db_record('aut', recid)
+    except RecordGetterError:
+        wrong_pid_types = ['lit', 'dat', 'con', 'job', 'ins', 'exp', 'jou']
+        for current_pid_type in wrong_pid_types:
+            try:
+                wrong_data = get_db_record(current_pid_type, recid)
+                if wrong_data:
+                    abort(404)
+            except RecordGetterError:
+                pass
+
         try:
             url = get_legacy_url_for_recid(recid) + '/export/xm'
-            xml = requests.get(url)
-            record_regex = re.compile(
-                r"\<record\>.*\<\/record\>", re.MULTILINE + re.DOTALL)
-            xml_content = record_regex.search(xml.content).group()
-            data = marcxml2record(xml_content)
-            convert_for_form(data)
+            xml_response = requests.get(url)
+            xml_response.raise_for_status()
         except requests.exceptions.RequestException:
-            pass
-        data["control_number"] = recid
-    else:
-        return redirect(url_for("inspirehep_authors.new"))
+            if xml_response.status_code == 404:
+                abort(404)
+            abort(503)
+
+        xml = fromstring(xml_response.content)
+        xml_record = xml.xpath("//*[local-name() = 'record']")[0]
+        data = marcxml2record(tostring(xml_record))
```

Comment:
Sometimes we get responses like:
```
<collection xmlns="http://www.loc.gov/MARC21/slim">
<link/>
<script/>
<record>
<controlfield tag="001">4</controlfield>
<datafield tag="980" ind1=" " ind2=" ">
<subfield code="c">DELETED</subfield>
</datafield>
</record>
</collection>
```
When this happens, we still get redirected to the form. Some format check is needed to see if what we get is indeed an XML containing an author record.